src/common/result_code.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * Adyen API
+ *
+ * Contact: support@adyen.com
+ * Generated by: OpenAPI Generator (https://openapi-generator.tech)
+ */
+
+package common
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+type ResultCode int
+
+const (
+	AuthenticationFinished ResultCode = iota
+	AuthenticationNotRequired
+	Authorised
+	Cancelled
+	ChallengeShopper
+	Error
+	IdentifyShopper
+	Pending
+	PresentToShopper
+	Received
+	RedirectShopper
+	Refused
+)
+
+// ResultCodeValues should be in same order as the const defined above
+var ResultCodeValues = [...]string{
+	"AuthenticationFinished",
+	"AuthenticationNotRequired",
+	"Authorised",
+	"Cancelled",
+	"ChallengeShopper",
+	"Error",
+	"IdentifyShopper",
+	"Pending",
+	"PresentToShopper",
+	"Received",
+	"RedirectShopper",
+	"Refused",
+}
+
+func (s ResultCode) String() string {
+	return ResultCodeValues[s]
+}
+
+func ResultCodeFromString(value string) (ResultCode, error) {
+	for p, v := range ResultCodeValues {
+		if v == value {
+			return ResultCode(p), nil
+		}
+	}
+	return ResultCode(0), fmt.Errorf("ResultCode enum not found for given value")
+}
+
+// MarshalJSON marshals the enum as a quoted json string
+func (s *ResultCode) MarshalJSON() ([]byte, error) {
+	buffer := bytes.NewBufferString(`"`)
+	buffer.WriteString(s.String())
+	buffer.WriteString(`"`)
+	return buffer.Bytes(), nil
+}
+
+// UnmarshalJSON unmarshals a quoted json string to the enum value
+func (s *ResultCode) UnmarshalJSON(b []byte) error {
+	var j string
+	err := json.Unmarshal(b, &j)
+	if err != nil {
+		return err
+	}
+	// Note that if the string cannot be found then it will be set to the zero value, 'Created' in this case.
```

Comment:
I don't understand this comment. Seems like there is no enum for `Created`, if you want this logic then add `Created` ad first value in cont and array